tests/backup/backup_longevity_test.go

```diff
@@ -0,0 +1,409 @@
+package tests
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/portworx/torpedo/pkg/log"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	. "github.com/portworx/torpedo/tests"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	testTriggersConfigMap = "pxb-longevity-triggers"
+	configMapNS           = "default"
+	controlLoopSleepTime  = time.Second * 15
+)
+
+var (
+	// Stores mapping between chaos level and its freq. Values are hardcoded
+	triggerInterval map[string]map[int]time.Duration
+	// Stores which are disruptive triggers. When disruptive triggers are happening in test,
+	// other triggers are allowed to happen only after existing triggers are complete.
+	disruptiveTriggers map[string]bool
+
+	triggerFunctions     map[string]func(*[]*scheduler.Context, *chan *EventRecord)
+	emailTriggerFunction map[string]func()
+
+	// Pure Topology is disabled by default
+	pureTopologyEnabled = false
+
+	//Default is allow deploying apps both in storage and storageless nodes
+	hyperConvergedTypeEnabled = true
+
+	// Pure Topology Label array
+	labels []map[string]string
+)
+
+// TriggerFunction represents function signature of a testTrigger
+type TriggerFunction func(*[]*scheduler.Context, *chan *EventRecord)
+
+var _ = Describe("{BackupLongevity}", func() {
+
+	IsBackupLongevityRun = true
+
+	contexts := make([]*scheduler.Context, 0)
+	var triggerLock sync.Mutex
+	var emailTriggerLock sync.Mutex
+	var populateDone bool
+	triggerEventsChan := make(chan *EventRecord, 100)
+	triggerFunctions = map[string]func(*[]*scheduler.Context, *chan *EventRecord){
+		CreatePxBackup:           TriggerCreateBackup,
+		CreatePxBackupAndRestore: TriggerCreateBackupAndRestore,
+		CreateRandomRestore:      TriggerCreateRandomRestore,
+		DeployBackupApps:         TriggerDeployBackupApps,
+	}
+	//Creating a distinct trigger to make sure email triggers at regular intervals
+	emailTriggerFunction = map[string]func(){
+		EmailReporter: TriggerEmailReporter,
+	}
+
+	BeforeEach(func() {
+		if !populateDone {
+			StartPxBackupTorpedoTest("BackupLongevityTest",
+				"Longevity Run For Backup", nil, 0, ATrivedi, Q1FY24)
+			populateIntervals()
+			//  populateDisruptiveTriggers()
```

Comment:
delete if not required

Reply:
This may be required in future, for now only normal events are added but we are planning to add pxb disruptive events in next cycles.

Reply:
please add a ToDO comment